lib/metadata/chordMetadata.hpp

```diff
@@ -164,50 +176,43 @@ struct chordMetadata {
     }
 };
 
-inline void chord_metadata_init(chordMetadata* c) {
-    bzero(c, sizeof(chordMetadata));
+inline void chord_metadata_init(std::shared_ptr<chordMetadata> meta) {
+    // ??
+    *meta = chordMetadata();
```

Comment:
Ditto on the `??`. This code looks weird. We have a nice constructor that initializes the metadata, I don't think we need an explicitly initialization function.